### PR TITLE
feat: add OCI support for Argo CD sources

### PR DESCRIPTION
**Pull request by robinlieb**
## Description

This PR adds the capability to mutate OCI references in Argo CD resources and replaces the given OCI reference to the Zarf registry URL.

This change adds the possibility to use Manifest and Helm Charts in the repoURL in the Application, like explained in the [OCI Docs in Argo CD](https://argo-cd.readthedocs.io/en/stable/user-guide/oci/) with the first two examples. To archive that it also changes the patching of the URL in repository secrets to use the registry instead of the git server if oci is used.

Additionally, added the patching of the repo url for Argo CD [repo-creds](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#repository-credentials) secret type.

## Related Issue

Relates to #3046 

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/internal/agent/hooks/argocd-repository.go [hunk 1]
```diff
@@ -107,11 +135,20 @@ func mutateRepositorySecret(ctx context.Context, r *v1.AdmissionRequest, cluster
...
 
 // Patch updates of the Argo Repository Secret.
-func populateArgoRepositoryPatchOperations(repoURL string, gitServer state.GitServerInfo) []operations.PatchOperation {
+func populateArgoRepositoryPatchOperations(repoURL string, gitServer state.GitServerInfo, registryInfo state.RegistryInfo, isOCI bool) []operations.PatchOperation {
 	var patches []operations.PatchOperation
 	patches = append(patches, operations.ReplacePatchOperation("/data/url", base64.StdEncoding.EncodeToString([]byte(repoURL))))
-	patches = append(patches, operations.ReplacePatchOperation("/data/username", base64.StdEncoding.EncodeToString([]byte(gitServer.PullUsername))))
-	patches = append(patches, operations.ReplacePatchOperation("/data/password", base64.StdEncoding.EncodeToString([]byte(gitServer.PullPassword))))
+
+	if isOCI {
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-repository.go]: Nit: for code patterns used across zarf (of those replaced thus far) we have been upgrading to use [the happy path is left-aligned](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88) concept whereby - when possible we process conditions and return early and avoid else returns.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Made a first pass - spent a healthy amount of time educating myself on argo further. 

Left a nit that I think we can cleanup in a few places. Separately want to consider some additional scope for this change:
1. E2E testing - unit tests get us some good feedback loops on the internal dependency and behaviors but this is largely an external integration. I would love to think about how we test this is operational. 
2. Documentation - right now this can be appending to the existing tutorial. There may be more opportunity for a more granular pass at argocd orchestration docs. 

Two minor requests:
1. Can you comment on an issue if/when you pick it up for work (or at least with the PR) - In doing so we can assign you and help prevent any colliding submissions (as well as remember to track for reviews on the project board. 
2. Add context to the PR description. It took me a fair bit of reading to tell that the associated issue was asking for a feature around oci registries when you actually solved for oci images. The former isn't actually as valid  and you solved the problem correctly here IMO - but I had to hunt down the delta and process. 

Thank you for the continued support and contributions - I very much appreciate it!

**Comment by robinlieb** [comment]
@brandtkeller thanks for the initial review. I added changes which addresses all of the mentioned points.
Added two applications in the Argo CD examples showcasing the the use of Helm Charts and manifests from OCI.
Added E2E test on base of this example.
Also refactored the patch logic for Argo CD resources since this contained some duplication.

For the other two points, will definitely keep that in mind for future work.

Update: For the sake of proper documentation, should I create a new issue specifying the OCI support for manifests?

**Comment by al-jeyapal** [comment]
The issue we might have with this is that ArgoCD will try to hit the registry over HTTPS. We were getting this in the Argo logs:

```
Failed to load target state:
 failed to generate manifest for source 1 of 2:
   rpc error: code = Unknown desc = failed to resolve revision "0.0.0":
     cannot get digest for revision 0.0.0:
       Head "https://zarf-docker-registry.zarf.svc.cluster.local:5000/v2/helm-charts/manifests/0.0.0":
         http: server gave HTTP response to HTTPS client
```

Helm allows you to use http via:

```
 helm pull --plain-http oci://...
```

However it doesn't look like Argo exposes an option for that. It exposes a `--insecure-skip-server-verification` but that only allows it to trust unverified certs.

We're currently trying to work around this by putting an internal `caddy` service to act as a HTTPS proxy in front of the zarf registry (on #4327). Still working through the details.

The easier option would be to have a HTTPS endpoint exposed on the zarf registry (just internally as a service not as an ingress/GatewayAPI route), even if it just uses a self-signed certificate. Not sure if that's a route we want to take though.

**Comment by robinlieb** [comment]
The [failed pipeline](https://github.com/zarf-dev/zarf/actions/runs/20178960749) Is probably related to the version bump of Argo CD, which throws an error with the gitea.localhost setup in the test. Not sure yet what exactly changed in Argo CD but will investigate further on how to resolve that.

**Comment by robinlieb** [comment]
External test fails because `gitea.localhost` gets resolved to 127.0.0.1 when calling git / libcurl in Argo CD Repo Server. This behavior changed with Helm 7.4.0 / Argo CD v2.12.0, where Argo CD moved to ubuntu:24.04 base image, which is stricter / RFC 6761 conform in terms of localhost resolution.
Added `curloptResolve` option for git config in repo sever.
Alternative approach would move from `gitea.localhost` to e.g. `gitea.local` in tests, but that would require to add this entry in `/etc/hosts` on local machines running external tests and in pipeline.

**Comment by al-jeyapal** [comment]
Thanks for the work on this @robinlieb. Out of curiosity, how did you work around ArgoCD forcing Helm to use HTTPS when interacting with OCI?  We had to put a HTTPS proxy in place to work around that.

**Comment by robinlieb** [comment]
@al-jeyapal setting `insecureOCIForceHttp` to `true` for the repository secret should make that possible. Have you tried that? Had you also the chance to build and test the branch and could provide feedback if that world work on your end?

**Comment by al-jeyapal** [comment]
> @al-jeyapal setting `insecureOCIForceHttp` to `true` for the repository secret should make that possible. Have you tried that? Had you also the chance to build and test the branch and could provide feedback if that world work on your end?

Thanks @robinlieb I hadn't seen any references to that in the doco, but can see it [in the code](https://github.com/argoproj/argo-cd/blob/c4f3bb8be401e3feb2bf8a85334bd4998eebbbc3/pkg/apis/application/v1alpha1/repository_types.go#L117). Very helpful.

I'm hoping to be rotated back onto the project later this month, so will aim to give it a go when next on.

File: src/internal/agent/hooks/argocd-repository.go [hunk 2]
```diff
@@ -107,11 +101,29 @@ func mutateRepositorySecret(ctx context.Context, r *v1.AdmissionRequest, cluster
...
-	patches = append(patches, operations.ReplacePatchOperation("/data/password", base64.StdEncoding.EncodeToString([]byte(gitServer.PullPassword))))
+	patches = append(patches, operations.ReplacePatchOperation("/data/username", base64.StdEncoding.EncodeToString([]byte(username))))
+	patches = append(patches, operations.ReplacePatchOperation("/data/password", base64.StdEncoding.EncodeToString([]byte(password))))
+
+	if isOCIURL && registryInfo.IsInternal() && !useMTLS {
+		patches = append(patches, operations.ReplacePatchOperation("/data/insecureOCIForceHttp", base64.StdEncoding.EncodeToString([]byte("true"))))
+	}
+
+	if useMTLS && isOCIURL {
+		patches = append(patches, operations.ReplacePatchOperation("/data/tlsClientCertData", base64.StdEncoding.EncodeToString([]byte(cluster.RegistryClientTLSSecret))))
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-repository.go]: If I am following correctly - this is patching the `secret` tlsClientCertData with the constant `"zarf-registry-client-tls"` value. But I wonder if this is instead supposed to be the actual certificate data.... 

Documentation is a little ambiguous but the statement:
> tlsClientCertData and tlsClientCertKey refer to secrets where a TLS client certificate (tlsClientCertData) and the corresponding private key tlsClientCertKey are stored for accessing the repositories

from the `repository` perspective makes me think that from the `secret` perspective that we are mutating here that the data is not by reference/name but rather direct. 

I need to validate this further.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Making my way through each file - only one item stood out as a blocker thus far.

File: examples/argocd/baseline/values.yaml [hunk 3]
```diff
@@ -19,7 +19,45 @@ configs:
...
+# /etc/hosts entries.
+extraObjects:
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: git-config-localhost-resolve
+    data:
+      gitconfig: |
+        [http]
+            curloptResolve = gitea.localhost:3000:172.31.0.99
```
Comment (brandtkeller) [inline examples/argocd/baseline/values.yaml]: appreciate the callout on this change. Thinking out-loud: this has no effect on standard `zarf init` installations with in-cluster gitea as state will point to the gitea kubernetes service dns (which doesn't match gitea.localhost). 

I'm not opposed to keeping this here but I would be interested in any options for keeping it within the scope of tests.

Comment (brandtkeller) [reply examples/argocd/baseline/values.yaml]: @robinlieb How would you feel about trying to implement this addition as a deploy-time values using the `values` feature?

Comment (robinlieb) [reply examples/argocd/baseline/values.yaml]: Changed it to use deploy-time values.

File: src/internal/agent/hooks/argocd-repository.go (hunk 2 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-repository.go]: Updated it that both cert and key is used in the Argo CD Repo secret. Updated the test accordingly.

File: src/internal/agent/hooks/argocd-application.go [hunk 4]
```diff
@@ -118,6 +129,82 @@ func getPatchedRepoURL(ctx context.Context, repoURL string, gs state.GitServerIn
...
+		return repoURL, nil
+	}
+
+	if isOCIURL {
+		return mutateOCIURL(ctx, repoURL, registryAddress, isPatchedClusterIP)
+	}
+	return mutateGitURL(ctx, repoURL, gs)
+}
+
+func shouldMutateURL(operation v1.Operation, isOCIURL bool, repoURL, registryAddress, clusterIP string, gs state.GitServerInfo) (bool, bool, error) {
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-application.go]: Should we unit test and and doccomments for this function? three return values is slightly confusing without any supporting context. I also want to ensure we capture edge cases.

File: src/internal/agent/hooks/argocd-repository.go [hunk 5]
```diff
@@ -107,11 +103,30 @@ func mutateRepositorySecret(ctx context.Context, r *v1.AdmissionRequest, cluster
...
+
+	if useMTLS && isOCIURL {
+		patches = append(patches, operations.ReplacePatchOperation("/data/tlsClientCertData", cert.Cert))
+		patches = append(patches, operations.ReplacePatchOperation("/data/tlsClientCertKey", cert.Key))
+	}
 
 	return patches
 }
+
+// Helper for getting eiher git server of registry creds
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-repository.go]:
```suggestion
// Helper for getting either git server of registry creds
```

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Minor changes - running this through local testing now to understand the implications.

File: src/internal/agent/hooks/argocd-application.go (hunk 4 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-application.go]: Added docs and unit tests.

File: src/internal/agent/hooks/argocd-repository.go (hunk 5 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-repository.go]: Added change.

File: examples/argocd/zarf.yaml [hunk 6]
```diff
@@ -8,18 +8,24 @@ components:
...
-        version: 5.54.0
+        version: 9.0.5
         namespace: argocd
         url: https://argoproj.github.io/argo-helm
         releaseName: argocd-baseline
         valuesFiles:
           - baseline/values.yaml
+        # Required for 'src/test/external' to validate external git server.
+        values:
+          - sourcePath: ".extraObjects"
```
Comment (brandtkeller) [inline examples/argocd/zarf.yaml:19]: Deployment of the package will fail if you simply `zarf package create examples/argocd` now unless you provide the required values files on deploy. 

I think the strategy as it stands is sound for adding the testing information at-deploy - but we should include a default values file in the package which has these values specified (empty or defualt) so that the package can still deploy without deploy-time values.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
I believe tests may be failing related to the comment I made on the argocd package manifest itself. 

Otherwise I'd request one other change (once the argocd package is updated) which would be to run a package create/deploy under the `src/test/proxy/proxy_test.go` in `Test_3_OCIOpsPackage()` such that we continue to test argo for the proxy suite. 

I tested locally - it looks good to me and I can approve if all is good after updates.

**Comment by brandtkeller** [comment]
Recent PR's touched cluster state - if you can update with main then I'll evaluate any additional testing on airgapped clusters and advocate to get this approved/merged. Thanks for the patience.

**Comment by robinlieb** [comment]
> Recent PR's touched cluster state - if you can update with main then I'll evaluate any additional testing on airgapped clusters and advocate to get this approved/merged. Thanks for the patience.

Rebased.

File: src/internal/agent/hooks/argocd-application.go
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-application.go]: Looking at the new components - I believe we're enabling the orchestration of applications with argocd using only OCI - is that correct?

We made this change recently for an early return when the git server is not configured (for git repo mutations). Now that we have OCI do we need to check for scheme earlier and then determine which service is required?

**Review by brandtkeller — COMMENTED** [review]
Recent change I want to surface - apologies for the churn but this shouldn't be too difficult of a change.

File: src/internal/agent/hooks/argocd-application.go
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-application.go]: Good catch, thanks!
Updated that to check for both to have an early return, then check the repoURL for each individual source in the Application, in case there are mixed sources.

**Comment by zeljkobekcic** [comment]
Hey, I would really like to use this feature. Can I expect this feature to be in the next release?

**Comment by brandtkeller** [comment]
> Hey, I would really like to use this feature. Can I expect this feature to be in the next release?

In review! Next release is tomorrow - so if not that release then the following. 

@robinlieb Can you update with main before I run CI? We had some changes that will likely make them fail if I run now. 

Review nearly complete.

**Comment by robinlieb** [comment]
> > Hey, I would really like to use this feature. Can I expect this feature to be in the next release?
> 
> In review! Next release is tomorrow - so if not that release then the following.
> 
> @robinlieb Can you update with main before I run CI? We had some changes that will likely make them fail if I run now.
> 
> Review nearly complete.

@brandtkeller rebased.

File: src/internal/agent/hooks/argocd-repository.go
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-repository.go]: Doing some testing - Do we need to handle registry-only (OCI) for this resource?

File: src/internal/agent/hooks/argocd-appproject.go
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-appproject.go]: Are there OCI implications that we need to consider here as well?

**Review by brandtkeller — CHANGES_REQUESTED** [review]
I think there are tests implicitly passing with the argo package because we also have the git server configured (required for the package to fully deploy) and the corresponding logic can handle OCI - but when you disable the git server and remove the git-server related elements from the package it does not deploy the applications as expected.

File: src/internal/agent/hooks/argocd-repository.go
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-repository.go]: Unified the check to conform with applications. Added tests for that.

File: src/internal/agent/hooks/argocd-appproject.go
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-appproject.go]: Unified the check to conform with applications. Added tests for that.

File: src/internal/agent/hooks/argocd-application.go [hunk 7]
```diff
@@ -122,6 +142,95 @@ func getPatchedRepoURL(ctx context.Context, repoURL string, gs state.GitServerIn
...
+// shouldMutateURL reports whether the given repoURL should be mutated for the admission operation.
+//
+// shouldMutate is true for Create operations and for Update operations where repoURL has not
+// already been patched to the Zarf-managed registry or git server address.
+//
+// isPatchedClusterIP is true when repoURL already matches the registry cluster-IP address
+func shouldMutateURL(operation v1.Operation, isOCIURL bool, repoURL, registryAddress, clusterIP string, gs state.GitServerInfo) (shouldMutate bool, isPatchedClusterIP bool, err error) {
+	isCreate := operation == v1.Create
+	isUpdate := operation == v1.Update
+	if isCreate {
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-application.go]: We removed checks for the create operation #4691, instead reaching for idempotent behaviors. There were edgecases regarding pre-mutated urls on create. I would look to verify if this is required or if the opreation is idempotent regardless.

File: src/internal/agent/hooks/common.go [hunk 8]
```diff
@@ -20,6 +21,13 @@ import (
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
+const (
+	// AgentErrTransformGitURL is thrown when the agent fails to make the git url a Zarf compatible url
+	AgentErrTransformGitURL = "unable to transform the git url"
```
Comment (brandtkeller) [inline src/internal/agent/hooks/common.go:26]: nit: if we have these constants then they likeky qualify as good sentinel errors. I won't block on this.

File: src/internal/agent/hooks/argocd-appproject.go [hunk 9]
```diff
@@ -53,24 +53,32 @@ func mutateAppProject(ctx context.Context, r *v1.AdmissionRequest, cluster *clus
...
+	if err != nil {
+		return nil, err
+	}
 
 	patches := make([]operations.PatchOperation, 0)
 
 	for idx, repo := range proj.Spec.SourceRepos {
-		patchedURL, err := getPatchedRepoURL(ctx, repo, s.GitServer)
+		patchedURL, err := getPatchedRepoURL(ctx, repo, registryAddress, clusterIP, s.GitServer, r)
 		// The AppProject can also include source repositories like '*' (as in the default project),
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-appproject.go:84]: Something we should address in the near future is detecting wildcards before the patch operation. This current flow could open up quite a few avenues for unintended consequences on misconfiguration.

File: src/internal/agent/hooks/argocd-application.go [hunk 10]
```diff
@@ -58,31 +58,46 @@ func NewApplicationMutationHook(ctx context.Context, cluster *cluster.Cluster) o
...
+	registryAddress, clusterIP, err := cluster.GetServiceInfoFromRegistryAddress(ctx, s.RegistryInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	l.Info("mutating the ArgoCD Application",
 		"name", app.Name,
-		"operation", r.Operation,
-		"gitServer", s.GitServer.Address)
+		"operation", r.Operation)
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-application.go]: Minor nit - we lose the ability to track what the intended mutation was going to try - this has been useful in debugging historically.

File: src/internal/agent/hooks/argocd-application.go
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-application.go]: This check is missing the asymmetry for both git server address and registry address. shouldMutateURL also performs this check in a sense. So I think we just need to resolve how and where to best perform that validation.

File: src/internal/agent/hooks/argocd-applicationset.go [hunk 11]
```diff
@@ -64,25 +64,40 @@ func mutateApplicationSet(ctx context.Context, r *v1.AdmissionRequest, cluster *
...
+
+	if !anyZarfServiceUsable(requiresGit, requiresRegistry, s) {
+		l.Debug("no Zarf services configured for source URL schemes, skipping ArgoCD ApplicationSet mutation")
+		return &operations.Result{Allowed: true}, nil
+	}
+
+	l.Info("mutating the ArgoCD ApplicationSet",
 		"name", appSet.Name,
-		"gitServer", s.GitServer.Address)
+		"operation", r.Operation)
```
Comment (brandtkeller) [inline src/internal/agent/hooks/argocd-applicationset.go]: Same comment for losing address in the logs

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Some of my last comments (hopefully) - only other thing I was going to look at is where we landed for testing with the proxy init model and mTLS.

File: src/internal/agent/hooks/argocd-applicationset.go (hunk 11 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-applicationset.go]: Added both, git server and registry.

File: src/internal/agent/hooks/argocd-application.go (hunk 10 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-application.go]: Added both, git server and registry.

File: src/internal/agent/hooks/argocd-application.go (hunk 7 above)
Comment (robinlieb) [reply src/internal/agent/hooks/argocd-application.go]: Removed the operation checks.

File: src/test/e2e/22_git_and_gitops_test.go [hunk 12]
```diff
@@ -192,15 +192,25 @@ func waitFluxPodInfoDeployment(t *testing.T) {
 func waitArgoDeployment(t *testing.T) {
 	// Deploy the argocd example and verify that it works
 	tmpdir := t.TempDir()
-	stdOut, stdErr, err := e2e.Zarf(t, "package", "create", "examples/argocd", "-o", tmpdir, "--skip-sbom")
+	stdOut, stdErr, err := e2e.Zarf(t, "package", "create", "examples/argocd", "-o", tmpdir, "--skip-sbom", "--features=\"values=true\"")
```
Comment (brandtkeller) [inline src/test/e2e/22_git_and_gitops_test.go]: For sake of consistency and not introducing any weirdness with cobras stringtostring parser let's be consistent with the features arguments we use elsewhere
```suggestion
	stdOut, stdErr, err := e2e.Zarf(t, "package", "create", "examples/argocd", "-o", tmpdir, "--skip-sbom", "--features", "values=true")
```

File: src/test/e2e/22_git_and_gitops_test.go [hunk 13]
```diff
@@ -192,15 +192,25 @@ func waitFluxPodInfoDeployment(t *testing.T) {
...
 	require.NoError(t, err, stdOut, stdErr)
 	packageName := fmt.Sprintf("zarf-package-argocd-%s.tar.zst", e2e.Arch)
 	path := filepath.Join(tmpdir, packageName)
-	stdOut, stdErr, err = e2e.Zarf(t, "package", "deploy", path, "--components=argocd-apps", "--confirm")
+	stdOut, stdErr, err = e2e.Zarf(t, "package", "deploy", path, "--components=argocd-apps", "--confirm", "--features=\"values=true\"")
```
Comment (brandtkeller) [inline src/test/e2e/22_git_and_gitops_test.go]:
```suggestion
	stdOut, stdErr, err = e2e.Zarf(t, "package", "deploy", path, "--components=argocd-apps", "--confirm", "--features", "values=true")
```

File: src/test/external/ext_in_cluster_test.go [hunk 14]
```diff
@@ -107,7 +107,7 @@ func (suite *ExtInClusterTestSuite) Test_0_Mirror() {
 	// Use Zarf to mirror a package to the services (do this as test 0 so that the registry is unpolluted)
 	t := suite.T()
 	tmpdir := t.TempDir()
-	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom")
+	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom", "--features=\"values=true\"")
```
Comment (brandtkeller) [inline src/test/external/ext_in_cluster_test.go]:
```suggestion
	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom", "--features", "values=true")
```

File: src/test/external/ext_out_cluster_test.go [hunk 15]
```diff
@@ -110,7 +136,7 @@ func (suite *ExtOutClusterTestSuite) Test_0_Mirror() {
 	// Use Zarf to mirror a package to the services (do this as test 0 so that the registry is unpolluted)
 	t := suite.T()
 	tmpdir := t.TempDir()
-	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom")
+	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom", "--features=\"values=true\"")
```
Comment (brandtkeller) [inline src/test/external/ext_out_cluster_test.go]:
```suggestion
	err := exec.CmdWithPrint(zarfBinPath, "package", "create", "../../../examples/argocd", "-o", tmpdir, "--skip-sbom", "--features", "values=true")
```

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Minor changes - please rebase with main and I'll kickoff CI and approve/advocate for second review. Appreciate the patience again.

**Comment by robinlieb** [comment]
> Minor changes - please rebase with main and I'll kickoff CI and approve/advocate for second review. Appreciate the patience again.

Rebased and proposed changes applied, thanks!

**Review by brandtkeller — APPROVED** [review]

**Review by AustinAbro321 — APPROVED** [review]
LGTM, nice work
